test/parallel/test-net-sync-cork.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+
+const server = net.createServer(handle);
+
+const N = 100;
+const buf = Buffer.alloc(2, 'a');
+
+server.listen(common.PORT, function() {
+  const conn = net.connect(common.PORT);
+
+  conn.on('connect', () => {
+    let res = true;
+    let i = 0;
+    for (; i < N && res; i++) {
+      conn.cork();
+      conn.write(buf);
+      res = conn.write(buf);
+      conn.uncork();
+    }
+    assert.equal(i, N);
```

Comment:
_maybe_ also assert `res` is still true, since it could have been `false` in the last iteration.